### PR TITLE
[NETBEANS-5053] - cleanup Set raw type warnings..

### PR DESCRIPTION
**Pull request by BradWalker**
I am trying to cleanup a bunch of the Set raw type warnings.. They look like this..

   [repeat] /home/bwalker/src/netbeans/cpplite/cpplite.debugger/src/org/netbeans/modules/cpplite/debugger/breakpoints/CPPLiteBreakpointActionProvider.java:50: warning: [rawtypes] found raw type: Set
   [repeat]     private static final Set actions = Collections.singleton (
   [repeat]                          ^
   [repeat]   missing type arguments for generic class Set<E>
   [repeat]   where E is a type-variable:
   [repeat]     E extends Object declared in interface Set

This change reduces the warnings dramatically. The ones remaining are in public interfaces
and it's beyond the scope for me to tackle that.. Hopefully, in the future.

### CODE REVIEW

File: java/ant.debugger/src/org/netbeans/modules/ant/debugger/AntDebugger.java [hunk 1]
```diff
@@ -306,9 +306,8 @@ private void stopHere(AntEvent event) {
         updateUI();
         currentFile = event.getScriptLocation();
         // update variable values
-        Set properties = event.getPropertyNames ();
-        variables = (String[]) properties.toArray 
-            (new String [properties.size ()]);
+        Set<String> properties = event.getPropertyNames();
+        variables = (String[])properties.toArray(new String [properties.size()]);
```
Comment (mklaehn) [inline java/ant.debugger/src/org/netbeans/modules/ant/debugger/AntDebugger.java:310]: Redundant cast after change
```suggestion
        variables = properties.toArray(new String [properties.size()]);
```

File: ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java [hunk 2]
```diff
@@ -136,33 +136,33 @@ private static Map getElementDecls() {
...
-            Set charInstructions = new TreeSet(Arrays.asList(new String[]{"apply-templates", // NOI18N
+            Set<String> charInstructions = new TreeSet<>(Arrays.<String>asList(new String[]{"apply-templates", // NOI18N
             "call-template","apply-imports","for-each","value-of", // NOI18N
             "copy-of","number","choose","if","text","copy", // NOI18N
             "variable","message","fallback"})); // NOI18N
 
-            Set instructions = new TreeSet(charInstructions);
-            instructions.addAll(Arrays.asList(new String[]{"processing-instruction", // NOI18N
+            Set<String> instructions = new TreeSet<>(charInstructions);
+            instructions.addAll(Arrays.<String>asList(new String[]{"processing-instruction", // NOI18N
```
Comment (mklaehn) [inline ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java:150]: inferrable type informations
```suggestion
            instructions.addAll(Arrays.asList(new String[]{"processing-instruction", // NOI18N
```

strictly speaking the array creation is not needed as well

File: ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java [hunk 3]
```diff
@@ -136,33 +136,33 @@ private static Map getElementDecls() {
...
             Set emptySet = new TreeSet();
             String spaceAtt = "xml:space";  // NOI18N
-            Set tmpSet;
+            Set<String> tmpSet;
 
             ////////////////////////////////////////////////
             // Initialize common sets
 
-            Set charInstructions = new TreeSet(Arrays.asList(new String[]{"apply-templates", // NOI18N
+            Set<String> charInstructions = new TreeSet<>(Arrays.<String>asList(new String[]{"apply-templates", // NOI18N
```
Comment (mklaehn) [inline ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java:144]: inferrable type informations
```suggestion
            Set<String> charInstructions = new TreeSet<>(Arrays.asList(new String[]{"apply-templates", // NOI18N
```
strictly speaking the array creation is not needed as well

File: ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java [hunk 4]
```diff
@@ -136,33 +136,33 @@ private static Map getElementDecls() {
...
 
-            Set charTemplate = charInstructions; // We don't care about PCDATA
+            Set<String> charTemplate = charInstructions; // We don't care about PCDATA
 
-            template = new TreeSet(instructions);
+            template = new TreeSet<>(instructions);
             template.add(resultElements);
 
-            Set topLevel = new TreeSet(Arrays.asList(new String[]{"import","include","strip-space", // NOI18N
+            Set<String> topLevel = new TreeSet<>(Arrays.<String>asList(new String[]{"import","include","strip-space", // NOI18N
```
Comment (mklaehn) [inline ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java:158]: inferrable type informations
```suggestion
            Set<String> topLevel = new TreeSet<>(Arrays.asList(new String[]{"import","include","strip-space", // NOI18N
```
strictly speaking the array creation is not needed as well

File: ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java [hunk 5]
```diff
@@ -136,33 +136,33 @@ private static Map getElementDecls() {
...
             "preserve-space","output","key","decimal-format","attribute-set", // NOI18N
             "variable","param","template","namespace-alias"})); // NOI18N
 
-            Set topLevelAttr = new TreeSet(Arrays.asList(new String[]{"extension-element-prefixes", // NOI18N
+            Set<String> topLevelAttr = new TreeSet(Arrays.<String>asList(new String[]{"extension-element-prefixes", // NOI18N
```
Comment (mklaehn) [inline ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java:162]: inferrable type informations
```suggestion
            Set<String> topLevelAttr = new TreeSet(Arrays.asList(new String[]{"extension-element-prefixes", // NOI18N
```
strictly speaking the array creation is not needed as well

File: ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java [hunk 6]
```diff
@@ -136,33 +136,33 @@ private static Map getElementDecls() {
...
             "exclude-result-prefixes","id","version",spaceAtt})); // NOI18N
 
-            resultElementAttr = new TreeSet(Arrays.asList(new String[]{"extension-element-prefixes", // NOI18N
+            resultElementAttr = new TreeSet<>(Arrays.<String>asList(new String[]{"extension-element-prefixes", // NOI18N
```
Comment (mklaehn) [inline ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java:165]: inferrable type informations
```suggestion
            resultElementAttr = new TreeSet<>(Arrays.asList(new String[]{"extension-element-prefixes", // NOI18N
```
strictly speaking the array creation is not needed as well

File: ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java [hunk 7]
```diff
@@ -365,7 +365,7 @@ private static Set getTemplate() {
 
     private static Set getXslFunctions() {
         if (xslFunctions == null) {
-            xslFunctions = new TreeSet(Arrays.asList(new String[]{
+            xslFunctions = new TreeSet<>(Arrays.<String>asList(new String[]{
```
Comment (mklaehn) [inline ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java:368]: inferrable type informations
```suggestion
            xslFunctions = new TreeSet<>(Arrays.asList(new String[]{
```
strictly speaking the array creation is not needed as well

File: ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java [hunk 8]
```diff
@@ -378,7 +378,7 @@ private static Set getXslFunctions() {
 
     private static Set getXPathAxes() {
         if (xpathAxes == null) {
-            xpathAxes = new TreeSet(Arrays.asList(new String[]{"ancestor::", "ancestor-or-self::", // NOI18N
+            xpathAxes = new TreeSet<>(Arrays.<String>asList(new String[]{"ancestor::", "ancestor-or-self::", // NOI18N
```
Comment (mklaehn) [inline ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java:381]: inferrable type informations
```suggestion
            xpathAxes = new TreeSet<>(Arrays.asList(new String[]{"ancestor::", "ancestor-or-self::", // NOI18N
```
strictly speaking the array creation is not needed as well

**Review by mklaehn — COMMENTED** [review]
I like fewer warnings (rawtype and otherwise).

Commented some changes

File: java/ant.debugger/src/org/netbeans/modules/ant/debugger/AntDebugger.java (hunk 1 above)
Comment (BradWalker) [reply java/ant.debugger/src/org/netbeans/modules/ant/debugger/AntDebugger.java:310]: Fixed..

File: ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java (hunk 2 above)
Comment (BradWalker) [reply ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java:150]: changed..

File: ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java (hunk 3 above)
Comment (BradWalker) [reply ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java:144]: changed..

File: ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java (hunk 4 above)
Comment (BradWalker) [reply ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java:158]: changed..

File: ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java (hunk 5 above)
Comment (BradWalker) [reply ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java:162]: changed..

File: ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java (hunk 6 above)
Comment (BradWalker) [reply ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java:165]: changed..

File: ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java (hunk 8 above)
Comment (BradWalker) [reply ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java:381]: changed..

File: ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java (hunk 2 above)
Comment (BradWalker) [reply ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java:150]: changed..

File: ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java (hunk 4 above)
Comment (BradWalker) [reply ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java:158]: changed..

File: ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java (hunk 5 above)
Comment (BradWalker) [reply ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java:162]: changed..

File: ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java (hunk 6 above)
Comment (BradWalker) [reply ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java:165]: chnaged..

File: ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java (hunk 7 above)
Comment (BradWalker) [reply ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java:368]: changed..

File: ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java (hunk 8 above)
Comment (BradWalker) [reply ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java:381]: changed..

File: ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java (hunk 3 above)
Comment (BradWalker) [reply ide/xsl/src/org/netbeans/modules/xsl/grammar/XSLGrammarQuery.java:144]: changed..

**Comment by BradWalker** [comment]
Hey @mklaehn , I tried to also remove the array creation.. But, it gave me an error.. So I left it.. But, I did make the suggested changed you had..

Thanks very much for looking at this!!

**Review by mklaehn — APPROVED** [review]
nice work

**Comment by matthiasblaesing** [comment]
Thank you - lets get this in.

**Pull request merged**